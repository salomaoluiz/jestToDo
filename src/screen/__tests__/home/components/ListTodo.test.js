import ListTodo from "../../../home/components/listTodo";
import React from "react";
import { shallow } from "enzyme";

describe("=== SCREEN === HOME --- COMPONENTS -- ListTodo", () => {
  //#region Configurations

  const props = {
    listTodo: [
      {
        id: 1,
        text: "Configurar o Jest",
        selected: true
      },
      {
        id: 2,
        text: "Fazer os testes",
        selected: false
      }
    ],
    markTodo: jest.fn(),
    removeTodo: jest.fn()
  };
  const wrapper = shallow(<ListTodo {...props} />);

  //#endregion

  //#region Render and is Equal

  it("Render and is Equal", () => {
    expect(wrapper).toMatchSnapshot();
  });

  //#endregion

  //#region Find KeyExtractor and return the id Value

  it("Find KeyExtractor and return the id Value", () => {
    const keyExtractor = wrapper
      .find("FlatList")
      .props()
      .keyExtractor(props.listTodo[0]);

    expect(keyExtractor).toEqual("1");
  });

  //#endregion

  //#region Render Item when is selected

  it("Render Item when is selected ", () => {
    const itemMarkWrapper = wrapper
      .find("FlatList")
      .props()
      .renderItem({ item: props.listTodo[0] });
    expect(itemMarkWrapper).toMatchSnapshot();
  });

  //#endregion

  describe("Render Item -", () => {
    //#region Configurations

    const itemWrapper = shallow(
      wrapper
        .find("FlatList")
        .props()
        .renderItem({ item: props.listTodo[1] })
    );

    //#endregion

    //#region Render Item when is not selected

    it("Render Item when is not selected ", () => {
      expect(itemWrapper).toMatchSnapshot();
    });

    //#endregion

    //#region Find TouchableOpacity of the Finish Todo and press

    it("Find TouchableOpacity of the Finish Todo and press", () => {
      itemWrapper
        .find("TouchableOpacity")
        .at(0)
        .props()
        .onPress();

      expect(props.markTodo).toHaveBeenCalledTimes(1);
      expect(wrapper.state().refresh).toBe(true);
    });

    //#endregion

    //#region Find TouchableOpacity of the delete Todo press

    it("Find TouchableOpacity of the delete Todo press", () => {
      itemWrapper
        .find("TouchableOpacity")
        .at(1)
        .props()
        .onPress();

      expect(props.removeTodo).toHaveBeenCalledTimes(1);
      expect(wrapper.state().refresh).toBe(false);
    });

    //#endregion
  });
});
