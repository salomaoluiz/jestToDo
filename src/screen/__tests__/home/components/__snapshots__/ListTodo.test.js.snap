// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`=== SCREEN === HOME --- COMPONENTS -- ListTodo Render Item - Render Item when is not selected  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "margin": 10,
      "width": 200,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#000"
      name="checkbox-blank-outline"
      size={20}
    />
    <Text
      style={
        Array [
          Object {
            "fontSize": 24,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Fazer os testes
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Icon
      allowFontScaling={false}
      color="#000"
      name="delete"
      size={20}
    />
  </TouchableOpacity>
</View>
`;

exports[`=== SCREEN === HOME --- COMPONENTS -- ListTodo Render Item when is selected  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "margin": 10,
      "width": 200,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#000"
      name="checkbox-marked"
      size={20}
    />
    <Text
      style={
        Array [
          Object {
            "fontSize": 24,
            "textAlign": "center",
          },
          Object {
            "color": "#cecece",
            "textDecorationLine": "line-through",
          },
        ]
      }
    >
      Configurar o Jest
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Icon
      allowFontScaling={false}
      color="#000"
      name="delete"
      size={20}
    />
  </TouchableOpacity>
</View>
`;

exports[`=== SCREEN === HOME --- COMPONENTS -- ListTodo Render and is Equal 1`] = `
<FlatList
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  refreshing={false}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
